Shopify/cli
# Themes cli migration

### WHY are these changes introduced?
CLI2 is going to be deprecated in some weeks. Although the team is working on the migration of the `themes` commands from Ruby to Typescript, a fallback is needed in case those task don't finish before the deadline. 

Another disadvantage of having both CLI version on different repositories is that each time a new feature of fix is needed, two PRs must be created and the deployment process should be synced between them.

So with this PR the CLI2 is going to be embed inside the CLI3. As a first step the complete CLI2 will be copied inside but in a next step only the code related with `theme` commands should be remain


### WHAT is this pull request doing?
- Move the Ruby CLI2 codebase to the `theme` module
- Integrate CLI3 `themes` commands with the embed version of the CLI2
- Add a feature flag to switch between the using of the `embed` version and the `bundled` one
- Integrate lint tasks in the current github workflow



### MISSING points
- `theme-cli` unit tests are not working on github. Removed from the workflow
- `theme-cli` includes the complete CLI codebase. Retain only `theme commands` related code.

### How to test your changes?
- `embed` version of the CLI2will be used in the development environment as this env variable `SHOPIFY_CLI_THEME_CLI=embed`  has been added to the `dev.yml`, so please run `dev up ` before the first running 
- Run any theme command using `--verbose` flag and check that the message `Using embed Theme CLI` is being displayed
- In the release version the `bundled` version of the CLI2 is going to be used so far. To simulate this in the local environment run any `theme` command prepending this `SHOPIFY_CLI_THEME_CLI=bundled`


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
